registry/server/appRoutes/routes/updateAppRoute.ts

```diff
@@ -0,0 +1,74 @@
+import {
+    Request,
+    Response,
+} from 'express';
+import Joi from '@hapi/joi';
+import _ from 'lodash/fp';
+
+import db from '../../db';
+import validateRequestFactory from '../../common/services/validateRequest';
+import {
+    prepareAppRouteToRespond,
+} from '../services/prepareAppRoute';
+import {
+    stringifyJSON,
+} from '../../common/services/json';
+import {
+    partialAppRouteSchema,
+} from '../interfaces';
+import {
+    appRouteIdSchema,
+} from '../interfaces';
+
+type UpdateAppRouteRequestParams = {
+    id: string
+};
+
+const validateRequestBeforeUpdateAppRoute = validateRequestFactory([
+    {
+        schema: Joi.object({
+            id: appRouteIdSchema.required(),
+        }),
+        selector: _.get('params'),
+    },
+    {
+        schema: partialAppRouteSchema,
+        selector: _.get('body')
+    },
+]);
+
+const updateAppRoute = async (req: Request<UpdateAppRouteRequestParams>, res: Response) => {
+    await validateRequestBeforeUpdateAppRoute(req, res);
+
+    const {
+        slots: appRouteSlots,
+        ...appRoute
+    } = req.body;
+    const appRouteId = req.params.id;
+
+    await db('routes').where('id', appRouteId).update(appRoute);
+
+    if (!_.isEmpty(appRouteSlots)) {
+        await db('route_slots').where('routeId', appRouteId).delete();
```

Comment:
batchInsert & delete should be a part of the same transaction/query

Reply:
Same goes to `db('routes').where('id', appRouteId).update(appRoute);`

Reply:
Done